torch/_export/pass_infra/proxy_value.py

```diff
@@ -1,13 +1,15 @@
-# mypy: allow-untyped-defs
 # pyre-strict
 from typing import Union
-
+from typing import Iterator
 import torch
 
+from typing import TypeVar
+
+_T = TypeVar("_T")
 
 class ProxyValue:
     # pyre-ignore
-    def __init__(self, data, proxy: Union[torch.fx.Proxy, torch.fx.Node]):
+    def __init__(self, data: Union[Iterator[_T], torch.Tensor], proxy: Union[torch.fx.Proxy, torch.fx.Node]):
```

Comment:
```suggestion
    def __init__(self, data: Union[Iterable[_T], torch.Tensor], proxy: Union[torch.fx.Proxy, torch.fx.Node]):
```

Reply:
Alas still no dice:

```
>>> Lint for torch/_export/pass_infra/proxy_value.py:

  Error (MYPY) [misc]
    Incompatible types in "yield from" (actual type "_T@__init__", expected
    type "_T@__iter__")

        38  |
        39  |    # pyre-ignore
        40  |    def __iter__(self) -> Iterator[_T]:
    >>> 41  |        yield from self.data
        42  |
        43  |    def __bool__(self) -> bool:
        44  |        return bool(self.data)
```

Feel free to commandeer this PR from me and add me as a reviewer. If you figure out how to do this, you should get the credit :P

Reply:
This is because the `_T` in `__init__` and the `_T` in `__iter__` represent different `_T` values. If you want them to be related then the class needs to be `Generic[_T]`.

Reply:
TIL. Thanks @aorenste!